### PR TITLE
[JBEAP-27261] On Windows run prospero using temporary jboss-modules to avoid resource locking

### PR DESCRIPTION
**Pull request by spyrkob**
Upstream: #706
Issue: https://issues.redhat.com/browse/JBEAP-27261

### CODE REVIEW

File: dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat [hunk 1]
```diff
@@ -187,6 +187,13 @@ if exist "%PROSPERO_HOME%\jboss-modules.jar" (
   goto END
 )
 
+rem Use a copy of jboss-modules to avoid locking issues when jboss-modules is updated
+:COPY_JBOSS_MODULES
+    set "TMP_JBOSS_MODULES=%PROSPERO_HOME%\jboss-modules~%RANDOM%.tmp"
+    rem loop while we find a non-existing filename
+    if exist "%TMP_JBOSS_MODULES%" goto :COPY_JBOSS_MODULES
+    cp "%RUNJAR%" "%TMP_JBOSS_MODULES%"
```
Comment (yersan) [inline dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat:195]: This `cp` should be replaced with `copy`

**Review by yersan — CHANGES_REQUESTED** [review]

**Comment by yersan** [comment]
@spyrkob what about these lines: 

https://github.com/wildfly-extras/prospero/blob/c0505f56a792cb15b8ae9786d014fc2c8f4b678e/dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat#L72-L74

It looks like there is a `)` that is doing nothing there?

File: dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat [hunk 2]
```diff
@@ -252,3 +259,6 @@ if %errorlevel% equ 10 (
 if "x%NOPAUSE%" == "x" pause
 
 :END_NO_PAUSE
+set EXIT_LEVEL="%errorlevel%"
+if exist "%TMP_JBOSS_MODULES%" rm -rf "%TMP_JBOSS_MODULES%"
```
Comment (yersan) [inline dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat:263]: `rm -rf "%TMP_JBOSS_MODULES%"` should be `del /Q "%TMP_JBOSS_MODULES%"

**Review by yersan — CHANGES_REQUESTED** [review]

**Comment by spyrkob** [comment]
Thanks @yersan, updated now

**Review by yersan — APPROVED** [review]

**Review by yersan — COMMENTED** [review]
@spyrkob I approved this, but let me do additional testsing and verify it when the server is it installed as a service

**Review by yersan — APPROVED** [review]
Looks good to me

**Pull request merged**